core/src/main/java/io/questdb/griffin/AlterCommandExecution.java

```diff
@@ -0,0 +1,235 @@
+        }
+    }
+
+    // Executes alter command
+    // If writer is busy exception is thrown
+    static void executeAlterStatementSyncOrFail(
+            CairoEngine engine,
+            AlterStatement alterStatement,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        if (alterStatement != null) {
+            assert alterStatement.getTableName() != null;
+            try (
+                    TableWriter writer = engine.getWriter(
+                            sqlExecutionContext.getCairoSecurityContext(),
+                            alterStatement.getTableName(), "alter statement execution"
+                    )
+            ) {
+                alterStatement.apply(writer, true);
+            } catch (TableStructureChangesException e) {
+                assert false : "TableStructureChangesException not happen when acceptStructureChange passed as true";
+            }
+        }
+    }
+
+    /***
+     * Sets up execution wait sequence to listen to the Engine async writer events
+     * @param engine Cairo Engine to consume events from
+     * @param sequence sequence to subscribe ot the events
+     */
+    public static void setUpEngineAsyncWriterEventWait(CairoEngine engine, SCSequence sequence) {
+        final FanOut writerEventFanOut = engine.getMessageBus().getTableWriterEventFanOut();
+        writerEventFanOut.and(sequence);
+    }
+
+    /***
+     * Cleans up execution wait sequence to listen to the Engine async writer events
+     * @param engine Cairo Engine subscribed to
+     * @param sequence to unsubscribe from Writer Events
+     */
+    public static void stopEngineAsyncWriterEventWait(CairoEngine engine, SCSequence sequence) {
+        engine.getMessageBus().getTableWriterEventFanOut().remove(sequence);
+        sequence.clear();
+    }
+
+    /***
+     * Blocking wait for the Writer Event
+     * @param engine Cairo Engine
+     * @param commandId command to wait the reply to
+     * @param responseConsumeSequence consume sequence to await async response from Table Writer
+     * @param writerAsyncCommandBusyWaitTimeout maximum wait timeout, microseconds
+     * @param queryTableNamePosition table name position in alter SQL query
+     * @return null if success, or instance of SqlException of wait is timed out or resulted in an error
+     */
+    public static SqlException waitWriterEvent(
+            CairoEngine engine,
+            long commandId,
+            Sequence responseConsumeSequence,
+            long writerAsyncCommandBusyWaitTimeout,
+            int queryTableNamePosition
+    ) {
+        final MicrosecondClock clock = engine.getConfiguration().getMicrosecondClock();
+        final long start = clock.getTicks();
+        final RingQueue<TableWriterTask> tableWriterEventQueue = engine.getMessageBus().getTableWriterEventQueue();
+
+        while (true) {
+            long seq = responseConsumeSequence.next();
```

Comment:
could we align name of this sequence to `eventSubSeq` throughout?

Reply:
removed class